server/server.go

```diff
@@ -202,108 +70,18 @@ type server struct {
-		return
-	}
-
-	// Init engines from -stores.
-	engines, err := initEngines(*stores)
-	if err != nil {
-		log.Errorf("Failed to initialize engines from -stores=%s: %v", *stores, err)
-		return
-	}
-	if len(engines) == 0 {
-		log.Errorf("No valid engines specified after initializing from -stores=%s", *stores)
-		return
-	}
-
-	err = s.start(engines, *attrs, *httpAddr, false)
-	defer s.stop()
-	if err != nil {
-		log.Errorf("Cockroach server exited with error: %v", err)
-		return
-	}
-
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill)
-
-	// Block until one of the signals above is received.
-	<-c
-}
-
-// parseAttributes parses a colon-separated list of strings,
-// filtering empty strings (i.e. ",," will yield no attributes.
-// Returns the list of strings as Attributes.
-func parseAttributes(attrsStr string) proto.Attributes {
-	var filtered []string
-	for _, attr := range strings.Split(attrsStr, ":") {
-		if len(attr) != 0 {
-			filtered = append(filtered, attr)
-		}
-	}
-	sort.Strings(filtered)
-	return proto.Attributes{Attrs: filtered}
-}
-
-// initEngines interprets the stores parameter to initialize a slice of
-// engine.Engine objects.
-func initEngines(stores string) ([]engine.Engine, error) {
-	// Error if regexp doesn't match.
-	storeSpecs := storesRE.FindAllStringSubmatch(stores, -1)
-	if storeSpecs == nil || len(storeSpecs) == 0 {
-		return nil, util.Errorf("invalid or empty engines specification %q", stores)
-	}
-
-	engines := []engine.Engine{}
-	for _, store := range storeSpecs {
-		if len(store) != 4 {
-			return nil, util.Errorf("unable to parse attributes and path from store %q", store[0])
-		}
-		// There are two matches for each store specification: the colon-separated
-		// list of attributes and the path.
-		engine, err := initEngine(store[1], store[2])
-		if err != nil {
-			return nil, util.Errorf("unable to init engine for store %q: %v", store[0], err)
-		}
-		engines = append(engines, engine)
-	}
-	log.Infof("Initialized %d storage engine(s)", len(engines))
-
-	return engines, nil
-}
-
-// initEngine parses the store attributes as a colon-separated list
-// and instantiates an engine based on the dir parameter. If dir parses
-// to an integer, it's taken to mean an in-memory engine; otherwise,
-// dir is treated as a path and a RocksDB engine is created.
-func initEngine(attrsStr, path string) (engine.Engine, error) {
-	attrs := parseAttributes(attrsStr)
-	if size, err := strconv.ParseUint(path, 10, 64); err == nil {
-		if size == 0 {
-			return nil, util.Errorf("unable to initialize an in-memory store with capacity 0")
-		}
-		return engine.NewInMem(attrs, int64(size)), nil
-		// TODO(spencer): should be using rocksdb for in-memory stores and
-		// relegate the InMem engine to usage only from unittests.
+// NewServer creates a Server from a server.Context.
+func NewServer(ctx *Context) (*Server, error) {
+	if ctx == nil {
+		return nil, errors.New("NewServer: ctx must not be null.")
```

Comment:
We use our homegrown util.Errorf for errors. This obviates the need to put in prefixes like "NewServer:", as it actually cites the file and line number exactly. Further, we don't start or end errors with capitalization and punctuation. In this case use:

    return nil, util.Errorf("ctx must not be null")